### PR TITLE
Fix for unless requisite when pip is not installed

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Fix for #56131

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

File: tests/integration/modules/test_state.py [hunk 1]
```diff
@@ -2268,3 +2268,27 @@ def test_state_sls_lazyloader_allows_recursion(self):
...
                          'Success!')
         self.assertTrue(state_run[state_id]['result'])
+
+    def test_issue_56131(self):
+        module_path = os.path.join(RUNTIME_VARS.CODE_DIR, 'pip.py')
+        if six.PY3:
+            modulec_path = os.path.join(RUNTIME_VARS.CODE_DIR, '__pycache__', 'pip.pyc')
+        else:
+            modulec_path = os.path.join(RUNTIME_VARS.CODE_DIR, 'pip.pyc')
+        print('*' * 80)
```
Comment (Ch3LL) [inline tests/integration/modules/test_state.py]: do we want to keep these prints here?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Comment by max-arnold** [comment]
@dwoz I believe this line should start with `if` (not `or`), otherwise it will iterate and delete all modules:

https://github.com/saltstack/salt/blob/1746e6ac60d1b6f9f2d3cb372e043d2aad66fcc0/salt/states/pip_state.py#L54

**Comment by dwoz** [comment]
> @dwoz I believe this line should start with `if` (not `or`), otherwise it will iterate and delete all modules:
> 
> https://github.com/saltstack/salt/blob/1746e6ac60d1b6f9f2d3cb372e043d2aad66fcc0/salt/states/pip_state.py#L54

Good catch, thanks!

File: tests/integration/modules/test_state.py (hunk 1 above)
Comment (dwoz) [reply tests/integration/modules/test_state.py]: Nope, will resolve this

**Review by Ch3LL — APPROVED** [review]

**Comment by Ch3LL** [comment]
looks like there are some lint failures that need to be cleaned up

**Review by DmitryKuzmenko — APPROVED** [review]

**Review by waynew — APPROVED** [review]

**Pull request merged**